lib/util/extensions/miq-uri.rb

```diff
@@ -0,0 +1,64 @@
+require 'uri'
+
+# Temporary workaround until https://github.com/ruby/ruby/pull/765 is
+# readily available.  As of this writing, ruby 2.0.0p643 and ruby 2.2.0 has this fix
+# and doesn't need this hack.  Ruby 2.1.5, and ruby 2.0.0p598 and lower don't have
+# this fix and NEED this hack.
+# Ruby 2.0: https://bugs.ruby-lang.org/issues/10873 (backported and in p643)
+# Ruby 2.1: https://bugs.ruby-lang.org/issues/10875 (backported and not released yet)
+# Ruby 2.2: https://bugs.ruby-lang.org/issues/10874 (backported and in 2.2.0)
+#
+# This HACK supports IPv6 literal address "host" values in URI::Generic.build.
+#
+# Before this hack:
+#  uri = URI::HTTPS.build(:path => "/sdk") # don't include :host => "::1" as it barfs
+#  uri.hostname = "::1"
+#  uri.to_s # => "https://[::1]/sdk"
+#
+# After this hack:
+#  uri = URI::HTTPS.build(:host => "::1", :path => "/sdk")
+#  uri.to_s # => "https://[::1]/sdk"
+#
+if RUBY_VERSION.start_with?("2.1") || (RUBY_VERSION == "2.0.0" && RUBY_PATCHLEVEL < 643)
```

Comment:
Rather than doing explicit version checking, we could do:

```ruby
def supports_ipv6_on_build?
  URI::HTTPS.build(:host => "::1", :path => "/sdk")
  true
rescue URI::InvalidComponentError
  false
end

if supports_ipv6_on_build?
  # etc
```

Then we don't have to update this when the 2.1 backport lands.